src/constants/notificationSchema.js

```diff
@@ -0,0 +1,76 @@
+export const notificationSchema = {
+  collaboration_manager_create: {
+    titleId: 'COLLABORATION_ESTABLISHED_BY_USER_MANAGER',
+    displayViewMoreInfoButton: 'true',
+    notificationMessage: 'A_COLLABORATION_WAS_CREATED_ON_YOUR_BEHALF',
+    moreDetailedDescription:
+      'A_COLLABORATION_WAS_CREATED_ON_YOUR_BEHALF_MORE_DETAILED',
+    availableButtons: [],
+  },
+  collaboration_request: {
+    titleId: 'COLLABORATION_REQUEST',
+    displayViewMoreInfoButton: 'true',
+    notificationMessage: 'COLLABORATION_VIEW_REQUEST_BRIEF',
+    moreDetailedDescription: 'COLLABORATION_VIEW_REQUEST_DESCRIPTION',
+    availableButtons: ['grant', 'decline'],
+    path: '/view_permission',
+  },
+  collaboration_approved: {
+    titleId: 'COLLABORATION_APPROVED_TITLE',
+    displayViewMoreInfoButton: 'none',
+    notificationMessage: 'COLLABORATION_APPROVED',
+    moreDetailedDescription: 'COLLABORATION_APPROVED',
+    availableButtons: ['revoke'],
```

Comment:
it's weird to define the list of possible buttons here, but have view be treated separately above. my recommendation would be to handle all the buttons together and come up with some sort of schema for defining them. for example 

```
availableButtons: [{
  id: 'approve-collaboration',
  labelId: 'APPROVE',
  display: 'primary',
  href: null,  // buttons would have links or onClick actions, but not both 
  onClick: async (notification) => ....
}]

```

Reply:
Ok I understand this one and the value of having all of the button prop logic live in notificationSchema.js.

However. Let's take the 'view' as an example. In NotificationsPane.jsx, I currently have:

```
{availableButtons.includes('view') && (
                    <Button
                      onClick={async () => {
                        setActiveCollaborationNotification(
                          notification,
                        );
                        await markRead(get(notification, 'guid'));
                        refreshNotifications();
                      }}
                      style={{ maxHeight: 40 }}
                      id="VIEW"
                    />
                  )}
```

Handling setActiveCollaborationNotification and refreshNotifications in notificationsSchema seems like it will require some gymnastics.

Similar issue for accessing onClose (a function passed into NotificationDetailsDialog as a prop) for 'decline' and 'grant'.

What do you think? More generally this is something I struggle with a lot: there's where I think the logic of these buttons, etc.  _ought_  to be written (in this case, arguably notificationSchema?) and then there's where _react seems to like them written_ (in this case, in the individual component files... e.g., NotificationDetailsDialog.jsx and NotificationsPane.jsx).

Reply:
Yeah i see where you're coming from. But the existing code will get messy once we have non-collaboration notifications and need to do something different instead of `setActiveCollaborationNotification`. Maybe the best solution is to do something similar to the cell renderer PR that just went in (or the way we handle fields) and define a separate notification component for each notification type.

Reply:
Yeah, that's kinda along the lines of what I was thinking, too. In this case, I will pivot to DEX-719 to get a better handle on that concept.